source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -140,12 +151,21 @@ mx::DocumentPtr Graph::loadDocument(mx::FilePath filename)
     mx::DocumentPtr doc = mx::createDocument();
     try
     {
-        mx::readFromXmlFile(doc, materialFilename, _searchPath, &readOptions);
+        if (!filename.isEmpty())
+        {
+            mx::readFromXmlFile(doc, filename, _searchPath, &readOptions);
+            std::string message;
+            if (!doc->validate(&message))
```

Comment:
This is just migrated from the loadDocument on the renderer which has been removed now as it's not used anymore.